### DISCUSSION TITLE
Reordering fields doesn't change the values

### DISCUSSION THREAD

**Discussion by rilwis** [react-hook-form/react-hook-form#3780] [Q&A]
I have 2 inputs: `f1` and `f2`. When I move them up/down, the DOM changes, but the values returned *always* in the order `f1-f2`.

Here is the Codesandbox:

https://codesandbox.io/s/react-hook-form-reorder-fields-mjg3k?file=/src/App.js

I always think the order of fields depends on the order of inputs, but I'm wrong.

Is there a way to synchronize the orders?

Thanks

**Comment by bluebill1049** [comment]
I think this is what you after: https://react-hook-form.com/api#useFieldArray

**Reply by rilwis** [reply]
I suppose it supports array of the same inputs? In my real case, my forms has different types of inputs: text, select, checkbox list, etc.

This is the real app: https://metabox.io/online-generator/

When adding fields from the list on the left, users can move them up/down. However, when submitting the form, the data doesn't have the correct order of the fields.

**Reply by aai-riyaj** [reply]
Do we have any updates here?
